examples/kubernetes-config/kubernetes-clustersubnet.json

```diff
@@ -14,7 +14,7 @@
     },
     "agentPoolProfiles": [
       {
-        "name": "agent",
+        "name": "agentpool1",
```

Comment:
It's actually the filename change that is breaking E2E configuration.

`examples/custom-pod-cidr/kubernetes.json --> examples/kubernetes-config/kubernetes-clustersubnet.json`

Do we mean to move the `examples/custom-pod-cidr/kubernetes.json` file around?